### ISSUE TITLE
[Youtube] RegexNotFoundError('Unable to extract Initial JS player n function name ...

### ISSUE THREAD

**Issue by luisgonzalez-moviments-net** [ytdl-org/youtube-dl#30712]
## Checklist

- [ ] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [ ] I've searched the bugtracker for similar bug reports including closed ones
- [ ] I've read bugs section in FAQ

## Verbose log

```
PASTE VERBOSE LOG HERE
ThinkPad-E570:~$ youtube-dl --verbose https://youtu.be/304guCcmt5w
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', 'https://youtu.be/304guCcmt5w']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.8.10 (CPython) - Linux-5.4.0-99-generic-x86_64-with-glibc2.29
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] 304guCcmt5w: Downloading webpage
[youtube] 304guCcmt5w: Downloading player 0abde7de
ERROR: 304guCcmt5w: Traceback (most recent call last):
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1434, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1407, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1394, in _extract_n_function_name
    return self._search_regex(
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
 (caused by RegexNotFoundError('Unable to extract \x1b[0;34mInitial JS player n function name\x1b[0m; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1434, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1407, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1394, in _extract_n_function_name
    return self._search_regex(
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1434, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1407, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1394, in _extract_n_function_name
    return self._search_regex(
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1866, in _real_extract
    self._unthrottle_format_urls(video_id, player_url, formats)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1451, in _unthrottle_format_urls
    n_response = self._n_descramble(n_param, player_url, video_id)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1441, in _n_descramble
    raise ExtractorError(traceback.format_exc(), cause=e, video_id=video_id)
youtube_dl.utils.ExtractorError: 304guCcmt5w: Traceback (most recent call last):
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1434, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1407, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/youtube.py", line 1394, in _extract_n_function_name
    return self._search_regex(
  File "/home/luis/.local/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
 (caused by RegexNotFoundError('Unable to extract \x1b[0;34mInitial JS player n function name\x1b[0m; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
ThinkPad-E570:~$

```

## Description

WRITE DESCRIPTION HERE
failing commands:
youtube-dl https://youtu.be/304guCcmt5w
youtube-dl -F https://youtu.be/304guCcmt5w
youtube-dl -f18 https://youtu.be/304guCcmt5w

with similar problem

**Comment by 89z** [comment]
Works fine:

~~~
PS C:\> youtube-dl.exe --version
2021.12.17

PS C:\> youtube-dl https://youtu.be/304guCcmt5w
[youtube] 304guCcmt5w: Downloading webpage
[download] Resuming download at byte 404099
[download] Destination: Conversando sobre Ucrania con Manolo Monereo-304guCcmt5w.f137.mp4
[download]   0.3% of 177.94MiB at 80.74KiB/s ETA 37:31
~~~

**Comment by dirkf** [comment]
OP didn't reveal that the yt-dl version is a later one taken from the git master. YT changed the player JS again but OP has not updated with the latest yt-dl master.

**Comment by 89z** [comment]
I dont get it, just comparing the output, we have the exact same version:

~~~
[debug] youtube-dl version 2021.12.17
~~~

With other languages, a "version" or "tag" is tied to a single commit, is that not the case here?

**Comment by dirkf** [comment]
If you hack your installation around, the version can stay the same. If you actually run a module from a git repository (eg, `python -m youtube_dl` in the yt-dl root directory), there's a `Git HEAD` line appended. No idea what happens when a commit is installed with _pip_.

**Comment by kokosowy** [comment]
I've just had the same issue. Just reinstall the module.
```
sudo -H pip uninstall youtube-dl
sudo -H pip install --upgrade youtube-dl
```

**Comment by 89z** [comment]
https://github.com/ytdl-org/youtube-dl/issues/30856

**Comment by 89z** [comment]
Looks like this is broken again, on master even

**Comment by abin2017** [comment]
i meet the same issue on the newest code on master branch

**Comment by dirkf** [comment]
Fixed again in a0068bd6bec16008bda7a39caecccbf84881c603.
